lib/anony/cops/define_deletion_strategy.rb

```diff
@@ -36,17 +36,16 @@ def on_class(node)
         end
 
         def model?(node)
-          return unless (superclass = node.children[1])
-
-          superclass.const_name == model_superclass_name
+          superclass = node.children[1]
```

Comment:
Did you consider looking for all leaf nodes that include `ApplicationRecord` or `ActiveRecord::Base` in their hierarchy instead of just looking at immediate children?

Reply:
So unfortunately it's not this straightforward because this is not a Ruby object, it's an AST node. As far as we could see this only has the context of the current file, so you'd have to do a bunch of finnagling to be able to look at the class hierarchy outside of that context.

Reply:
+1 , happy to have another look into some ways around this if blocking though  :~)

Reply:
I think it's probably fine. I guess we could potentially add a spec in payments-service to ensure all classes that inherit from ApplicationRecord and have children of their own are included in the `ModelSuperclass` list, but I think as @benk-gc says there's a limit to what we can check from a cop without a lot of extra work.

Maybe something that can be looked into another time. "Good enough" is better than not improving things because they're not yet perfect after all :)